### DISCUSSION TITLE
The URL for connecting to PiKaraoke doesn't work

### DISCUSSION THREAD

**Discussion by Fr3d-P** [vicwomg/pikaraoke#297] [Q&A] [answered]
Hi all,

After installing PiKaraoke on a SD Card, all is working very well, except one little thing...
The url displayed for connecting to my raspberry is "http://raspberrypi:5555"
The barcode corresponds to this URL.
Obviously, this doesn't work, i can access the machine with http://raspberrypi.local:5555
What happened to "local"?
How can I change the Url advertised?

**Comment by RecursiveFunctions** [comment]
Try launching with  
`--prefer-ip`
Or 

`-u [url]`

**Reply by BentWrench** [reply]
That did it, thanks!

**Comment by vicwomg** [comment]
The hostname comes from the standard python library:
`socket.getfqdn()`

I've found that sometimes it returns .local at the end, and sometimes not. It's not clear to me why, but would be grateful if someone could look into this. However, in my testing either seemed to work fine. Perhaps this is not true in all network configurations, though. Again, I'm not a networking expert so I'm not sure why that might be the case.

As stated above, you can specify your own URL with the -u parameter, or have it skip hostnames altogether and use ip address with --prefer-ip

**Reply by RecursiveFunctions** [reply]
if it's not part of a domain, or if the system's DNS settings don't allow it to resolve the FQDN, `socket.getfqdn()` might append .local to the hostname.

On some systems, the function appends .local to the hostname if it can't find a fully qualified domain name (FQDN). This is determined by the
`/etc/hosts` file and DNS settings on your system.

https://www.howtogeek.com/167190/how-and-why-to-assign-the-.local-domain-to-your-raspberry-pi/

Your pi might also be using avahi, an open-source implementation of mDNS.

**Reply by BentWrench** [reply]
I had the same issue, and launching with --prefer-ip did the trick. I am using the Pi over WiFi, on my home network, and it just seems to take a long time for my other devices to be able to find my pi using the hostname, particularly my phone. So IP is a better option for me. Thanks for posting the work-around.

**Reply by vicwomg** [reply]
I'm curious to hear from more users if prefer-ip works better overall. If so, maybe that should be the default. It used to be that way, but it's a little less friendly looking to people who might prefer to type in the url manually. 

I think this hostname stuff all comes down to how your LAN and router are configured. Not sure what the secret sauce is to make it work.

**Reply by BentWrench** [reply]
The local LAN will use mDNS, also called Bonjour on a Mac

https://stevessmarthomeguide.com/multicast-dns/

It tends to work on my home LAN, but seems to take a while, for new devices. If it is a device that has connected before, it seems quicker. I haven't tested any of this, just my observation.

**Comment by Fr3d-P** [comment]
For me, it's the only pi I have that isn't referenced as xxx.local.
With @RecursiveFunctions propositio, I used --prefer-ip, which works fine (it's not a big deal to not have the name, it's the barcode that is used)!

**Comment by vicwomg** [comment, marked as answer]
FYI, the latest master of pikaraoke includes this change:

`--prefer-ip` has been deprecated, IP is now default. Now hostname is opt-in with: `--prefer-hostname`
